src/targets/gpu/compile_ops.cpp

```diff
@@ -185,8 +185,7 @@ struct compile_plan
             results.begin(), results.end(), std::back_inserter(times), [&](const auto& cr) {
                 if(not cr.has_value())
                     return std::numeric_limits<double>::max();
-                return time_op(*ctx, cr->replace.code_object, to_shapes(cr->ins->inputs()), 20)
-                    .first;
```

Comment:
So earlier it was taking `host_time` for comparsion ? was that incorrect ?

Reply:
Was it not accurate because `enable_perf_measurements()` was recording start and stop events back to back?

Reply:
> Was it not accurate because enable_perf_measurements() was recording start and stop events back to back?

No, it didnt measure the host_time with `enable_perf_measurements()` set. 

The problem is each host_time run includes the launch overhead, which might vary. Instead we run the kernel without syncing until the end to minimize the launch overhead due to a larger queue depth.